imagersite/imager_profile/handlers.py

```diff
@@ -0,0 +1,54 @@
+from .models import ImagerProfile
+from django.contrib.auth.models import User
+from django.dispatch import receiver
+from django.db.models.signals import post_save, post_delete
+
+
+@receiver(post_save, sender=User)
+def create_user_profile(sender, **kwargs):
+    """A User should always have an ImagerProfile"""
+    instance = kwargs.get('instance')
+    if not instance or kwargs.get('raw', False):
+        return
+    try:
+        instance.profile
+    except ImagerProfile.DoesNotExist:
+        instance.profile = ImagerProfile()
+        instance.profile.save()
+
+
+@receiver(post_save, sender=ImagerProfile)
+def create_user(sender, **kwargs):
+    """An ImagerProfile should always have a User"""
+    instance = kwargs.get('instance')
+    if not instance or kwargs.get('raw', False):
+        return
+    try:
+        instance.user
+    except User.DoesNotExist:
+        instance.user = User()
+        instance.user.save()
+
+
+@receiver(post_delete, sender=ImagerProfile)
+def rm_user_profile(sender, **kwargs):
+    """If an ImagerProfile is deleted, delete it's User too"""
+    instance = kwargs.get('instance')
+    if not instance:
+        return
+    try:
+        instance.user.delete()
+    except:
+        pass
```

Comment:
These sorts of bare except clauses are discouraged for a reason.  As it stands, if anything at all goes wrong with your handler, it will silently fail to remove the opposite object.  That's not desirable.  If you intend to deal with an error, catch it.  Otherwise, let it bubble up and break things so you know when something's wrong.

I get the feeling you probably put this here because you were getting a ping-pong effect between this signal handler and the `rm_user` one defined below.  You don't need both.

In Django, items that have a foreign key to other objects are considered "dependent" on that object.  If the instance of that class that they depend on is deleted, they will also be deleted, automatically.  This preserved database integrity, and the fact that Django handles it in Python is an artifact of support dbs like sqlite3 that do not do so automatically on their own.  

Because of this automated cleanup, you should not need to delete objects with foreign keys from the database when the objects they depend on are removed.